### ISSUE TITLE
[Bug] Impossible to focus nav bar items using keyboard navigation

### ISSUE THREAD

**Issue by sarayourfriend** [WordPress/openverse-frontend#116]
## Description
Using only keyboard navigation, it is not possible to focus the nav bar items. They're not tabbable elements so the browser skips over them and right onto the search box.

## Reproduction

1. Open the frontend
2. Using the `tab` button, attempt to focus the nav bar dropdowns like the About and Resources dropdowns.

## Expectation
The dropdowns and their individual items should be focusable using keyboard navigation.

## Environment

This isn't environment specific. I've confirmed it with and without VoiceOver in Safari and Firefox on macOS.

## Additional context
I'm not sure why this is happening as the nav elements are all just plain anchor elements, but something is causing them to not be focusable.

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

**Comment by sarayourfriend** [comment]
After some more investigation I found that part of the problem is that you need to enable tab navigation for all controls in macOS: https://www.scottohara.me/blog/2014/10/03/link-tabbing-firefox-osx.html

After following the instructions on that page I was able to use keyboard navigation to focus most of the previously unfocusable elements on the page (include pages with lots of links like the `/sources` page.

However, this doesn't fix the issue that the nav bar dropdowns are not focusable, thereby making it impossible to access the dropdown contents.

**Comment by obulat** [comment]
I didn't know about enabling tab navigation in macOS!

Openverse is based on [Bulma framework](https://bulma.io/documentation/). While it was a great way to start, it also introduced some problems with accessibility. 

Keyboard navigation in the nav bar was mentioned in [an issue](https://github.com/jgthms/bulma/issues/872). The problem is that an `<a>` link without an `href` attribute is not focusable. And the top items in the nav bar do not have an address to link to. So, the solution could be either to convert the nav bar items with dropdown into buttons([this site](https://www.perkins.org/) has an example of button dropdowns), or add `tabindex=0` to them so that they are focusable even without an `href`. 

For context, there is a discussion about accessibility in Bulma's repo: https://github.com/jgthms/bulma/issues/53
We probably have similar problems, too.

**Comment by sarayourfriend** [comment]
My understanding is that anchors without `href`s are an anti-pattern right? Buttons seem preferrable in that case (though maybe placeholder anchors are a thing too?)

The responses in that Bulma issue from the maintainers are really disappointing.

I think to make the dropdowns fully ARIA compliant we should use the menu role: https://www.w3.org/TR/wai-aria-1.1/#menu

**Comment by sarayourfriend** [comment]
I think the `menu` role is for anything that is a "dropdown"? At least based on this example, it seems to be exactly the same use case: https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-1/menubar-1.html
